Redesign, update to bootstrap 3.2

dLobatog opened the pull request:
Please give me feedback.

I'm keeping a server open at 2466df00.ngrok.com , if you want to review and leave your comments here. It might change as I might be editing code while you're reviewing.

domcleal commented:
I like the blue and new layout, less sure about the green/ochre type colour...

I don't know if you plan to tackle it, but we should look at putting in a documentation landing page or something in the near future.  The number of manuals and docs we have is quite large, and the menu doesn't work too well.  I'd also like to include some plugin docs (@stbenjam had a go), and it'd be easier this way.  (This sort of thing: http://fedoraproject.org/en_GB/join-fedora)

dLobatog commented:
That sounds like a good idea, I didn't like the 3k pages manual approach either but didn't want to break away from it in a redesign. This was just a Sunday morning project so I'll get back to it from time to time and implement that.

domcleal commented:
Ah okay, I'm talking less about the length of the manual (though that's certainly a topic) but more splitting up the documentation dropdown menu - linking to manuals, articles, security stuff, release notes separately and clearly.

Maybe some other docs format (docbook etc) would be better as it might be possible to generate single and multi-page versions, as different people prefer each.

dLobatog commented:
Right, I think we're more or less on the same page about what would make it better. I'm ok with Jekyll (Liquid) - I've never worked with docbook to be honest - , it's relatively easy to render single pages & multi page manuals, even on demand. 

The gradient on the logo looks a bit dated in my opinion, but my judgment is probably not the best on these topics. @mairin , (you created the logo right?) do you think the logo/text look good enough or they need polishing?

domcleal commented:
Btw, https://github.com/theforeman/foreman-graphics contains the SVG etc.

dLobatog committed:
Redesign, update to bootstrap 3.2

dLobatog commented:
Updated, I think only links, some copy on the homepage, and the videos page need to be fixed now. You can check how pages look like by going to the same URL if you can't find the link on the new site. Feedback appreciated

ohadlevy commented on _includes/features.html:
```diff
@@ -0,0 +1,64 @@
+<section class="features">
+	<div class="container">
```
nitpick: why 8 spaces?

ehelms commented:
Overall I like the changes, as they feel crisper. A few thoughts from playing around with it:

1) If you move the hardhat beside the FOREMAN you have the same effect but gain some vertical space that will reveal more of the features below and catch people's eyes potentially.
2) Can you make the 'Install' button the same yellow as the hardhat? That would reflect theme a bit better.
3) Due to the font, and the darkness of the blue, the navigation links look broken. Locally testing the use of a bold version of the font, I think it looked much cleaner.
4) For me, the crisscrossing background on pages like the introduction and documentation is distracting
5) I am finding the right-hand sub-navigation on pages not intuitive given that a majority of sites use left hand navigation and conventions exist for a reason.
6) For our type of project, the source link may not make the most sense to present first to users. Maybe just a single link that says 'Get Started' instead of install, since that implies you are going to go through getting everything installed and initially setup with potentially a basic scenario. I believe we want to: call them to action, show them what it can do for them, and how to get it.

dLobatog commented:
Updated, I put the navigation bars on the left (I thought putting it on the right gives more emphasis to the content as English is read left to right, see the bootstrap docs for instance) && changed the yellow color to the hardhat yellow.

The videos page now loads simply thumbnails, and puts iframes of the youtube embedded player only when you click on them, so it loads in 1 second :smile:

dLobatog commented:
The videos page is completely redone, it misses some videos though because of the update & some conflicts I was lazy to solve. 

Minor fixes scattered all around & a new font on the latest commit. Please review & criticize.

stbenjam commented:
I have the aesthetic sense of a rock, so, look at my comments in that context, but:
- The yellow "INSTALL" still looks too dark -- I would prefer it to be lighter, or not yellow at all.  The two buttons on the bottom of the page are nice shades (and also feel misplaced)
- Overall,  the whole page feels too dark? The blue isn't Foreman Blue, is it?
- I'm unsure about the checkered background, especially since the manual navigation are not transparent.  I think it'd be nicer just plain white.

Although our Katello Green is hotly contested, I really like Katello.org, and think the Foreman site could take a couple of design elements from it:
- the top-level bar with the integrated logo is nice, and saves space on the front page, and would  make Foreman visible on every page, not just the hat
- I much prefer a sidebar with the latest news, it's quite hidden on the footer of all the pages in this design

stbenjam commented:
But lots of good things here :+1: I like the plugin page, especially.

dLobatog commented:
@stbenjam @ehelms Thanks for the reviews. I have to let go of the checkered background - and hopefully get @kybaker to review it too. Latest updates (almost there!!!):
- videos page is rendered client side, adding a new video is just adding the ID, author, and title
- minor CSS fixes
- wrote some stuff on the front page to get rid of loren ipsum and changed the screenshots to actual screenshots (thoughts on that text appreciated)
- fixed all links

As far as I can see we're just missing:
- fixes from the reviews (thanks!)
- figure out how to show other versions of the manual (probably some dropdown)

mmoll commented on _layouts/homepage.html:
```diff
@@ -1,47 +1,28 @@
+		<hr />
+		<div class='row center'>
+				<a class="btn btn-primary" href="introduction.html">Learn more</a>
+				<a class="btn btn-info" href="manuals/1.7/quickstart_guide.html">Get started</a>
```
1.7? :)

kybaker commented:
Overall I like the direction, but I have a few things I would change:
1) The purple background is strange. It does not match any colors in the app or website. One of the blues from the app would be better.
2) The background pattern behind the content does not add anything. It is just visual noise which distracts from the content. 
3) There are 4 different color buttons on the homepage alone. Should condense the color palette.
4) There is too much content on the homepage. We should think about having the broadest message on the homepage and getting more detail as users drill in. I think we should eliminate some content. If you had Provision, Configure and Monitor on the homepage twice with different information for each. 
5) The images are too small to tell me anything. Do they need to be there?
6) Too many colors in the footer and they clash with each other.
7) The left hand nav on the quick-start page seems unnecessary.

dLobatog commented:
- Suggested palette by @kybaker  (just so I don't forget) - http://www.colourlovers.com/palette/416765/quiet_morning
- Possibly change the color of the headers to distinguish them from <p>

I'm keeping a server open at http://2466df00.ngrok.com , if you want to review and leave your comments here. Content might change as I might be editing code while you're reviewing.

stbenjam commented:
I like the lighter blue!

adamruzicka commented:
Looks great, but i have some comments:
1) There is no navigation indicator in the navbar on the quickstart page, but there is on other pages
2) I propose some changes to the navbar, the change from the homepage to any of the categories is a bit too much [1]. Wouldn't it look better and feel more consistent if the navbar was the same everywhere? Maybe with the forman helmet missing on the homepage since its already in the body?

[1] http://ibin.co/20fmUl2LYtgy

dLobatog commented:
Updated with a version picker for the manual and release notes, and addressed most of the concerns here. Ready to go?

domcleal commented:
1. We seem to have lost a link to the security advisories page, it's probably got to go in Documentation somewhere.
2. Homepage: replace Discovery on the homepage with a block about the API?
3. Homepage: put RBAC/LDAP block next to Audits, as they're related

dLobatog commented:
Repushed, with fixes for the points 1,2,3 made by @domcleal above

domcleal commented:
This needs rebasing on top of #376 so the right Salt manual is linked.  It also means we lose the links to per-version plugin manuals, any thoughts about that?

dLobatog commented:
Yes, I think that is the last bit missing, I'll probably display the versions of each plugin prior to displaying the manual.

dLobatog commented:
I've added a simple landing page for plugins, see http://2466df00.ngrok.com/plugins/foreman_salt/ as an example. Chef doesn't have one as it only has one version. Plugin owners can add videos, and other related material in these pages as they see fit.

I think that was the last absolutely necessary change, let me know if you need me to change anything or if it's good to :ship: !

mmoll commented:
Is a rebase required after #378 has been merged? Anyway, I think this is good to go, :ship: :shipit:

ohadlevy commented:
minor items from my tests:
- opening the api should open v2 by default ( at least for recent versions)
- consider adding some js for image viewing, otherwise you have to click
  back after every image.
- new section is a bit hidden, maybe we can move it to a side bar or
  highlight it somehow?
- the media list is pretty huge, maybe we should consider adding filter /
  tagging to it?

huge improvement overall thanks!

stbenjam commented:
> I've added a simple landing page for plugins, see http://2466df00.ngrok.com/plugins/foreman_salt/

Nice :+1:

dLobatog commented:
Thanks @ohadlevy , I've set api_v2 as the default, and now the pictures on the index open a modal. 
I agree about the media, but that could be done at any point, now it's usable and much faster than in theforeman.org as it just loads thumbnails. 

Either better organization (by topic?) or fuzzy search on media.html would be nice and I could tackle these at some point, but I'd not hold this PR from merging because of the media page. It's become very time consuming and tedious to rebase and include the latest changes upstream - html is very different and git does (rightly so) not provide a great diff of the content.

domcleal commented:
Thanks @eLobato, this rocks!
